### PR TITLE
Remove SyncTable & Refactor CodeConfig Types

### PR DESCRIPTION
**Pull request by evantahler**

### CODE REVIEW

File: core/src/classes/codeConfig.ts [hunk 1]
```diff
@@ -152,29 +215,32 @@ export function extractNonNullParts(
 }
 
 export function getAutoBootstrappedProperty(
-  sourceConfigObject: ConfigurationObject,
-  otherConfigObjects: ConfigurationObject[]
+  sourceConfigObject: AnyConfigurationObject,
```
Comment (pedroslopez) [inline core/src/classes/codeConfig.ts]: I think this can be `SourceConfigurationObject`

File: core/src/modules/configLoaders/index.ts [hunk 2]
```diff
@@ -204,67 +198,79 @@ export async function processConfigObjects(
...
-            configObject,
+            configObject as TeamMemberConfigurationObject,
             externallyValidate,
             validate
           );
           break;
         case "profile":
           ids = await loadProfile(configObject, externallyValidate, validate);
           break;
-        case "synctable":
```
Comment (pedroslopez) [inline core/src/modules/configLoaders/index.ts:252]: Nice! I barely even knew this existed 😛

**Review by pedroslopez — APPROVED** [review]
Looks good! We now have proper types for each config object 😄

File: core/src/models/Source.ts [hunk 3]
```diff
@@ -302,7 +305,10 @@ export class Source extends LoggedModel<Source> {
     return this.idIsDefault() ? ConfigWriter.generateId(this.name) : this.id;
   }
 
-  async getConfigObject() {
+  async getConfigObject(): Promise<
+    | SourceConfigurationObject
+    | [SourceConfigurationObject, ScheduleConfigurationObject]
+  > {
```
Comment (evantahler) [inline core/src/models/Source.ts:311]: This is one is pretty strange, but at least now the types will remind us of that!

**Comment by evantahler** [comment]
Rebased and types updated to support additions from https://github.com/grouparoo/grouparoo/pull/2087

**Pull request merged**